core/unit_test/view/TestViewCustomizationAccessorFromMapping.hpp

```diff
@@ -0,0 +1,215 @@
+            std::enable_if_t<std::is_constructible_v<
+                                 Kokkos::default_accessor<element_type>,
+                                 Kokkos::default_accessor<OtherElementType>>,
+                             int> = 0>
+  KOKKOS_FUNCTION constexpr TestAccessorStrided(
+      const TestAccessorStrided<OtherElementType, MemorySpace>& other) noexcept
+      : size(other.size), stride(other.stride) {}
+
+  KOKKOS_FUNCTION
+  TestAccessorStrided(const size_t val, const size_t s_val)
+      : size(val), stride(s_val) {}
+
+  KOKKOS_FUNCTION
+  constexpr reference access(
+#ifndef KOKKOS_ENABLE_OPENACC
+      const data_handle_type& p,
+#else
+      // FIXME OPENACC: illegal address when passing by reference
+      data_handle_type p,
+#endif
+      size_t i) const noexcept {
+    return BarRefStrided{(p.get() + i), size, stride};
+  }
+
+  KOKKOS_FUNCTION
+  constexpr typename offset_policy::data_handle_type offset(
+#ifndef KOKKOS_ENABLE_OPENACC
+      const data_handle_type& p,
+#else
+      // FIXME OPENACC: illegal address when passing by reference
+      data_handle_type p,
+#endif
+      size_t i) const noexcept {
+    return p + i;
+  }
+
+  size_t size{0lu};
+  size_t stride{0lu};
+};
+
+// Use the customization point to inject the custom accessor
+template <class LayoutType, class DeviceType, class MemoryTraits>
+constexpr auto customize_view_arguments(
+    Kokkos::Impl::ViewArguments<BarStrided, LayoutType, DeviceType,
+                                MemoryTraits>) {
+  return Kokkos::Impl::ViewCustomArguments<
+      size_t,
+      TestAccessorStrided<BarStrided, typename DeviceType::memory_space>>{};
+}
+
+template <class LayoutType, class DeviceType, class MemoryTraits>
+constexpr auto customize_view_arguments(
+    Kokkos::Impl::ViewArguments<const BarStrided, LayoutType, DeviceType,
+                                MemoryTraits>) {
+  return Kokkos::Impl::ViewCustomArguments<
+      size_t, TestAccessorStrided<const BarStrided,
+                                  typename DeviceType::memory_space>>{};
+}
+
+// Customization point to compute allocation sizes
+template <class MappingType, class ElementType, class MemorySpace>
+size_t allocation_size_from_mapping_and_accessor(
+    const MappingType& map,
+    const TestAccessorStrided<ElementType, MemorySpace>& acc) {
+  return map.required_span_size() * acc.size;
+}
+
+// Customization point to create accessor from AccessorArg and Mapping
+template <class ElementType, class MemorySpace, class MappingType>
+constexpr auto accessor_from_mapping_and_accessor_arg(
+    Kokkos::Impl::AccessorTypeTag<
+        TestAccessorStrided<ElementType, MemorySpace>>,
+    const MappingType& map, const Kokkos::Impl::AccessorArg_t& acc_arg) {
+  return TestAccessorStrided<ElementType, MemorySpace>{
+      acc_arg.value, map.required_span_size()};
+}
+}  // namespace Foo
+
+void test_accessor_from_mapping() {
+  size_t ext0   = 10;
+  size_t ext1   = 10;
```

Comment:
I would avoid using the same value for both extends.
```suggestion
  size_t ext0   = 10;
  size_t ext1   = 11;
```